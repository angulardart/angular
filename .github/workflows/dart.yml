# Created with package:mono_repo v5.0.2
name: Dart CI
on:
  push:
    branches:
      - master
      - google
  pull_request:
    branches:
      - master
      - google
defaults:
  run:
    shell: bash
env:
  PUB_ENVIRONMENT: bot.github

jobs:
  job_001:
    name: mono_repo self validate
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:stable"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: stable
      - id: checkout
        uses: actions/checkout@v2.3.4
      - name: mono_repo self validate
        run: dart pub global activate mono_repo 5.0.2
      - name: mono_repo self validate
        run: dart pub global run mono_repo generate --validate
  job_002:
    name: "analyze; PKG: angular_ast; `dart analyze`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast;commands:analyze_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_ast_pub_upgrade
        name: angular_ast; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_ast
        run: dart pub upgrade
      - name: angular_ast; dart analyze
        if: "always() && steps.angular_ast_pub_upgrade.conclusion == 'success'"
        working-directory: angular_ast
        run: dart analyze
  job_003:
    name: "analyze; PKGS: angular_compiler, examples/hello_world; `dart analyze --fatal-infos`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler-examples/hello_world;commands:analyze_1"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler-examples/hello_world
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_compiler_pub_upgrade
        name: angular_compiler; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart pub upgrade
      - name: "angular_compiler; dart analyze --fatal-infos"
        if: "always() && steps.angular_compiler_pub_upgrade.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart analyze --fatal-infos
      - id: examples_hello_world_pub_upgrade
        name: examples/hello_world; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: examples/hello_world
        run: dart pub upgrade
      - name: "examples/hello_world; dart analyze --fatal-infos"
        if: "always() && steps.examples_hello_world_pub_upgrade.conclusion == 'success'"
        working-directory: examples/hello_world
        run: dart analyze --fatal-infos
  job_004:
    name: "build; PKG: _tests; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: _tests_pub_upgrade
        name: _tests; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: _tests
        run: dart pub upgrade
      - name: "_tests; pub run build_runner build --fail-on-severe"
        if: "always() && steps._tests_pub_upgrade.conclusion == 'success'"
        working-directory: _tests
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_005:
    name: "build; PKG: angular_ast; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_ast_pub_upgrade
        name: angular_ast; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_ast
        run: dart pub upgrade
      - name: "angular_ast; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_ast_pub_upgrade.conclusion == 'success'"
        working-directory: angular_ast
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_006:
    name: "build; PKG: angular_compiler; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_compiler_pub_upgrade
        name: angular_compiler; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart pub upgrade
      - name: "angular_compiler; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_compiler_pub_upgrade.conclusion == 'success'"
        working-directory: angular_compiler
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_007:
    name: "build; PKG: angular_forms; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_forms_pub_upgrade
        name: angular_forms; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_forms
        run: dart pub upgrade
      - name: "angular_forms; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_forms_pub_upgrade.conclusion == 'success'"
        working-directory: angular_forms
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_008:
    name: "build; PKG: angular_router; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_router_pub_upgrade
        name: angular_router; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_router
        run: dart pub upgrade
      - name: "angular_router; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_router_pub_upgrade.conclusion == 'success'"
        working-directory: angular_router
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_009:
    name: "build; PKG: angular_test; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_test_pub_upgrade
        name: angular_test; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_test
        run: dart pub upgrade
      - name: "angular_test; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_test_pub_upgrade.conclusion == 'success'"
        working-directory: angular_test
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_010:
    name: "build; PKG: examples/hello_world; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:examples/hello_world;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:examples/hello_world
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: examples_hello_world_pub_upgrade
        name: examples/hello_world; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: examples/hello_world
        run: dart pub upgrade
      - name: "examples/hello_world; pub run build_runner build --fail-on-severe"
        if: "always() && steps.examples_hello_world_pub_upgrade.conclusion == 'success'"
        working-directory: examples/hello_world
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_011:
    name: "unit_test; PKG: _tests; `pub run build_runner test --fail-on-severe -- -P browser`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests;commands:command_2"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: _tests_pub_upgrade
        name: _tests; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: _tests
        run: dart pub upgrade
      - name: "_tests; pub run build_runner test --fail-on-severe -- -P browser"
        if: "always() && steps._tests_pub_upgrade.conclusion == 'success'"
        working-directory: _tests
        run: pub run build_runner test --fail-on-severe -- -P browser
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_012:
    name: "unit_test; PKG: _tests; `pub run test -P vm`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests;commands:command_1"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: _tests_pub_upgrade
        name: _tests; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: _tests
        run: dart pub upgrade
      - name: "_tests; pub run test -P vm"
        if: "always() && steps._tests_pub_upgrade.conclusion == 'success'"
        working-directory: _tests
        run: pub run test -P vm
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_013:
    name: "unit_test; PKG: angular_ast; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_ast_pub_upgrade
        name: angular_ast; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_ast
        run: dart pub upgrade
      - name: "angular_ast; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_ast_pub_upgrade.conclusion == 'success'"
        working-directory: angular_ast
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_014:
    name: "unit_test; PKG: angular_compiler; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_compiler_pub_upgrade
        name: angular_compiler; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart pub upgrade
      - name: "angular_compiler; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_compiler_pub_upgrade.conclusion == 'success'"
        working-directory: angular_compiler
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_015:
    name: "unit_test; PKG: angular_forms; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_forms_pub_upgrade
        name: angular_forms; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_forms
        run: dart pub upgrade
      - name: "angular_forms; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_forms_pub_upgrade.conclusion == 'success'"
        working-directory: angular_forms
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_016:
    name: "unit_test; PKG: angular_router; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_router_pub_upgrade
        name: angular_router; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_router
        run: dart pub upgrade
      - name: "angular_router; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_router_pub_upgrade.conclusion == 'success'"
        working-directory: angular_router
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_017:
    name: "unit_test; PKG: angular_test; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_test_pub_upgrade
        name: angular_test; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_test
        run: dart pub upgrade
      - name: "angular_test; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_test_pub_upgrade.conclusion == 'success'"
        working-directory: angular_test
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_018:
    name: Notify failure
    if: "(github.event_name == 'push' || github.event_name == 'schedule') && failure()"
    runs-on: ubuntu-latest
    steps:
      - run: |
          curl -H "Content-Type: application/json" -X POST -d \
            "{'text':'Build failed! ${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}'}" \
            "${CHAT_WEBHOOK_URL}"
        env:
          CHAT_WEBHOOK_URL: "${{ secrets.TEAM_CHAT_WEBHOOK_URL }}"
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
      - job_011
      - job_012
      - job_013
      - job_014
      - job_015
      - job_016
      - job_017
