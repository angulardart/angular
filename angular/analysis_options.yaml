# This is a copy of ../analysis_options.yaml, with implicit-casts: false, and an "exclude: ...".

analyzer:
  strong-mode:
    implicit-casts: false
  exclude:
    - "tools/analyzer_plugin/bin/plugin.dart"
  errors:
    unused_element: error
    unused_import: error
    unused_local_variable: error
    dead_code: error

    # Allow importing .template.dart files without an analyzer error.
    uri_has_not_been_generated: ignore

    # The noise of deprecations isn't worth the effort.
    deprecated_member_use_from_same_package: ignore
linter:
  rules:
    - avoid_empty_else
    - avoid_init_to_null
    - avoid_null_checks_in_equality_operators
    - await_only_futures
    - camel_case_types
    - cancel_subscriptions
    - control_flow_in_finally
    - directives_ordering
    - empty_catches
    - empty_constructor_bodies
    - empty_statements
    - hash_and_equals
    - iterable_contains_unrelated_type
    - library_names
    - library_prefixes
    - list_remove_unrelated_type
    - only_throw_errors
    - overridden_fields
    - package_api_docs
    - package_names
    - package_prefixed_library_names
    - prefer_final_fields
    - prefer_is_not_empty
    - slash_for_doc_comments
    - test_types_in_equals
    - throw_in_finally
    - type_init_formals
    - unnecessary_brace_in_string_interps
    - unrelated_type_equality_checks
    - valid_regexps
