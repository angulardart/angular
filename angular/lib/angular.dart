/// The primary library for the [AngularDart web framework][AngularDart].
///
/// Import this library as follows:
///
/// ```
/// import 'package:angular/angular.dart';
/// ```
///
/// For help using this library, see the AngularDart documentation:
///
/// * [AngularDart guide][]
/// * [AngularDart cheat sheet][cheatsheet]
///
/// [AngularDart]: https://webdev.dartlang.org/angular
/// [AngularDart guide]: https://webdev.dartlang.org/angular/guide
/// [cheatsheet]: https://webdev.dartlang.org/angular/cheatsheet

library angular;

export 'core.dart';
export 'src/common/common_directives.dart';
export 'src/common/directives.dart';
export 'src/common/pipes.dart';
export 'src/core/angular_entrypoint.dart' show AngularEntrypoint;
export 'src/core/application_tokens.dart' hide APP_ID_RANDOM_PROVIDER;
export 'src/platform/bootstrap.dart';
export 'src/platform/dom/dom_tokens.dart';
export 'src/platform/dom/events/event_manager.dart' show EventManagerPlugin;
