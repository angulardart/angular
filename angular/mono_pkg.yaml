dart:
- stable
- dev

stages:
- analyze:
  - dartanalyzer: --fatal-warnings .
