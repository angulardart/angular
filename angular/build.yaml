targets:
  $default:
    builders:
      angular:
        generate_for:
          exclude:
            - "lib/src/build.dart"
            - "lib/src/compiler/**"
            - "lib/src/source_gen/**"

builders:
  angular:
    import: "package:angular/src/build.dart"
    builder_factories:
      - templatePlaceholder
      - templateCompiler
      - stylesheetCompiler
    auto_apply: dependents
    applies_builders:
      - "angular|placeholder_cleanup"
      - "angular|component_source_cleanup"
    # See https://github.com/dart-lang/angular/issues/988.
    is_optional: true
    required_inputs:
      - ".css"
    build_extensions:
      .css:
        - ".css.dart"
        - ".css.shim.dart"
      .dart:
        - ".template.dart"
post_process_builders:
  placeholder_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "placeholderCleanup"
  component_source_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "componentSourceCleanup"
    defaults:
      generate_for: ["lib/**"]
      release_options:
        enabled: true
