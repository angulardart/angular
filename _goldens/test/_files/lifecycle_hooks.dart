import 'package:angular/angular.dart';

@Component(
  selector: 'lifecycle-hooks',
  directives: [
    EmptyComponent,
  ],
  template: r'''
    <empty-comp></empty-comp>
    <ng-content></ng-content>
  ''',
)
class MostLifecycleHooksComponent extends MostLifecycleHooksSuperclass {}

class MostLifecycleHooksSuperclass
    implements
        OnDestroy,
        OnInit,
        AfterContentChecked,
        AfterContentInit,
        AfterViewChecked,
        AfterViewInit {
  @Input()
  String input;

  @override
  ngAfterContentChecked() {}

  @override
  ngAfterContentInit() {}

  @override
  ngAfterViewChecked() {}

  @override
  ngAfterViewInit() {}

  @override
  ngOnDestroy() {}

  @override
  ngOnInit() {}
}

@Component(
  selector: 'empty-comp',
  template: '',
)
class EmptyComponent {}

@Directive(
  selector: 'do-check',
)
class DoCheckDirective implements DoCheck {
  @override
  void ngDoCheck() {}

  @Input()
  String input;
}

@Component(
  selector: 'after-changes',
  template: '',
)
class AfterChangesComponent implements AfterChanges {
  @override
  void ngAfterChanges() {}

  @Input()
  String input;
}

@Component(
  selector: 'uses-lifecylce-hooks',
  directives: [
    MostLifecycleHooksComponent,
    EmptyComponent,
    DoCheckDirective,
    AfterChangesComponent,
  ],
  template: r'''
    <lifecycle-hooks [input]="input">
      <empty-comp></empty-comp>
    </lifecycle-hooks>
    <do-check [input]="input"></do-check>
    <after-changes [input]="input"></after-changes>
  ''',
)
class UsesLifecycleHooksComponent {
  var input = 'Hello World';
}
