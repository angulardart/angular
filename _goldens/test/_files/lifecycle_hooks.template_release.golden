// **************************************************************************
// Generator: Instance of 'Compiler'
// **************************************************************************

// ignore_for_file: cancel_subscriptions,constant_identifier_names,duplicate_import,non_constant_identifier_names,library_prefixes,UNUSED_IMPORT,UNUSED_SHOWN_NAME
import 'lifecycle_hooks.dart';
export 'lifecycle_hooks.dart';
import 'package:angular/angular.dart';
import 'package:angular/src/di/reflector.dart' as _ngRef;
import 'package:angular/angular.template.dart' as _ref0;
import 'package:angular/src/core/linker/app_view.dart';
import 'lifecycle_hooks.dart' as import1;
import 'package:angular/src/core/render/api.dart';
import 'package:angular/src/core/linker/view_type.dart' as import3;
import 'package:angular/src/core/change_detection/change_detection.dart';
import 'dart:html' as import5;
import 'package:angular/src/core/linker/app_view_utils.dart' as import6;
import 'package:angular/angular.dart';

const List<dynamic> styles$LifecycleHooksComponent = const [];

class ViewLifecycleHooksComponent0 extends AppView<import1.LifecycleHooksComponent> {
  static RenderComponentType _renderType;
  ViewLifecycleHooksComponent0(AppView<dynamic> parentView, int parentIndex) : super(import3.ViewType.COMPONENT, {}, parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    rootEl = import5.document.createElement('lifecycle-hooks');
    _renderType ??= import6.appViewUtils.createRenderType('', ViewEncapsulation.None, styles$LifecycleHooksComponent);
    setupComponentType(_renderType);
  }
  @override
  ComponentRef<import1.LifecycleHooksComponent> build() {
    final _rootEl = rootEl;
    final import5.HtmlElement parentRenderNode = initViewRoot(_rootEl);
    init(const [], null);
    return null;
  }
}

AppView<import1.LifecycleHooksComponent> viewFactory_LifecycleHooksComponent0(AppView<dynamic> parentView, int parentIndex) {
  return new ViewLifecycleHooksComponent0(parentView, parentIndex);
}

const List<dynamic> styles$LifecycleHooksComponentHost = const [];

class _ViewLifecycleHooksComponentHost0 extends AppView<dynamic> {
  ViewLifecycleHooksComponent0 _compView_0;
  import1.LifecycleHooksComponent _LifecycleHooksComponent_0_5;
  _ViewLifecycleHooksComponentHost0(AppView<dynamic> parentView, int parentIndex) : super(import3.ViewType.HOST, {}, parentView, parentIndex, ChangeDetectionStrategy.CheckAlways);
  @override
  ComponentRef build() {
    _compView_0 = new ViewLifecycleHooksComponent0(this, 0);
    rootEl = _compView_0.rootEl;
    _LifecycleHooksComponent_0_5 = new import1.LifecycleHooksComponent();
    _compView_0.create(_LifecycleHooksComponent_0_5, projectableNodes);
    init0(rootEl);
    return new ComponentRef<import1.LifecycleHooksComponent>(0, this, rootEl, _LifecycleHooksComponent_0_5);
  }

  @override
  void detectChangesInternal() {
    bool firstCheck = (this.cdState == 0);
    if (firstCheck) {
      _LifecycleHooksComponent_0_5.ngOnInit();
    }
    _LifecycleHooksComponent_0_5.ngDoCheck();
    if (firstCheck) {
      _LifecycleHooksComponent_0_5.ngAfterContentInit();
    }
    _LifecycleHooksComponent_0_5.ngAfterContentChecked();
    _compView_0.detectChanges();
    if (firstCheck) {
      _LifecycleHooksComponent_0_5.ngAfterViewInit();
    }
    _LifecycleHooksComponent_0_5.ngAfterViewChecked();
  }

  @override
  void destroyInternal() {
    _compView_0?.destroy();
    _LifecycleHooksComponent_0_5.ngOnDestroy();
  }
}

AppView viewFactory_LifecycleHooksComponentHost0(AppView<dynamic> parentView, int parentIndex) {
  return new _ViewLifecycleHooksComponentHost0(parentView, parentIndex);
}

const ComponentFactory<import1.LifecycleHooksComponent> LifecycleHooksComponentNgFactory = const ComponentFactory<import1.LifecycleHooksComponent>('lifecycle-hooks', viewFactory_LifecycleHooksComponentHost0, _LifecycleHooksComponentMetadata);
const _LifecycleHooksComponentMetadata = const [];
var _visited = false;
void initReflector() {
  if (_visited) {
    return;
  }
  _visited = true;

  _ngRef.registerComponent(LifecycleHooksComponent, LifecycleHooksComponentNgFactory);
  _ref0.initReflector();
}
